smslibrary/src/main/java/com/eis/communication/Payload.java

```diff
@@ -0,0 +1,61 @@
+package com.eis.communication;
+
+import com.eis.communication.exception.InvalidPayloadException;
+
+/**
+ * Payload abstract class
+ * @author Mattia Fanan
+ *
+ * @param T Payload data data-type
+ */
+public abstract class Payload<T> {
+    protected T data;
+
+    /**
+     * Create a payload
+     * @param data Data of the Payload
```

Comment:
Don't repeat yourself on param specifications
do something like `@param data of the payload`